### DISCUSSION TITLE
Easiest way to force a refresh every x milliseconds

### DISCUSSION THREAD

**Discussion by AyashiNoCeres** [emilk/egui#342] [Q&A] [answered]
Current the window only updates when it needs to, which is great but what about being able to override this?

What's the simplest way to do this? Say I want to force the screen to refresh every x milliseconds

**Comment by emilk** [comment]
There is no simple way to do this in eframe at the moment. What you can do is clone the [`RepaintSignal`](https://docs.rs/epi/0.11.0/epi/struct.Frame.html#method.repaint_signal) and send it to a background thread which just calls it every x milliseconds. Not great, not terrible.

**Reply by AyashiNoCeres** [reply]
thanks i'll give this a try!

**Comment by follower** [comment, marked as answer]
It wasn't clear to me if you specifically wanted a refresh/repaint that was _not_ every frame--but if "a refresh/repaint every frame " _is_ okay then you can use `request_repaint()` in your `update()` method, as used in some of the demos, e.g.:

https://github.com/emilk/egui/blob/0b52813f62de5a46dfde4c891c5f2f169b0aa8e1/egui_demo_lib/src/apps/demo/multi_touch.rs#L136-L137

and 

https://github.com/emilk/egui/blob/b1559963bf1c28877c6f04b7966d0614fced5e9d/egui_demo_lib/src/wrap_app.rs#L158-L247

But if a refresh _every_ frame is too frequent then you'd want to use the approach @emilk suggested.

Some docs here: https://docs.rs/egui/0.11.0/egui/struct.Context.html#method.request_repaint

**Reply by AyashiNoCeres** [reply]
Refreshing every frame is fine. so yeah that works pretty well for me! Thank you!

**Reply by nalply** [reply]
Sorry to ask, but what's the use case for repainting every frame? If the update doesn't work like you expect perhaps it's better to report this as a bug?

I respectfully ask this so you don't try to solve an [XY problem](https://en.wikipedia.org/wiki/XY_problem).
